NitroxClient/GameLogic/ChatUI/PlayerChatManager.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.MonoBehaviours.Gui.Chat;
+using NitroxModel.Logger;
+using NitroxModel.Packets;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace NitroxClient.GameLogic.ChatUI
+{
+    public class PlayerChatManager
+    {
+        public PlayerChatManager(IMultiplayerSession multiplayerSession)
+        {
+            this.multiplayerSession = multiplayerSession;
+            Player.main.StartCoroutine(LoadChatLogAsset());
+        }
+
+        private PlayerChat playerChat;
+        public Transform PlayerChaTransform => playerChat.transform;
+        private readonly IMultiplayerSession multiplayerSession;
+        private static readonly string assemblyPath = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+
+        private bool finishedLoading;
+        private bool showChatAfterLoading;
+        private bool selectChatAfterLoading;
+        private readonly List<ChatLogEntry> entriesBeforeFinishedLoading = new List<ChatLogEntry>();
```

Comment:
This could be refactored away to use `entries`. When the chat finishes loading it can add the entries up to that point and continue using entries for all new messages.